### PR TITLE
Release preparations

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Review by roxblnfk — APPROVED** [review]

**Comment by arogachev** [comment]
Thanks, but it's not ready for review yet (work in progress).

**Review by samdark — APPROVED** [review]

File: .github/workflows/composer-require-checker.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+name: Composer require checker
+
+jobs:
+  composer-require-checker:
+    uses: yiisoft/actions/.github/workflows/composer-require-checker.yml@master
+    with:
+      os: >-
+        ['ubuntu-latest']
+      php: >-
+        ['8.3']
```
Comment (vjik) [inline .github/workflows/composer-require-checker.yml:34]:
```suggestion
        ['8.1', '8.2', '8.3']
```

File: .github/workflows/bc.yml_ [hunk 2]
```diff
@@ -1,15 +1,14 @@
 on:
-    - pull_request
-    - push
+  - pull_request
+  - push
```
Comment (vjik) [inline .github/workflows/bc.yml_:3]: pull_request:
    paths-ignore:
      - 'docs/**'
      - 'README.md'
      - 'CHANGELOG.md'
      - '.gitignore'
      - '.gitattributes'
      - 'infection.json.dist'
      - 'phpunit.xml.dist'
      - 'psalm.xml'
  push:
    branches: ['master']
    paths-ignore:
      - 'docs/**'
      - 'README.md'
      - 'CHANGELOG.md'
      - '.gitignore'
      - '.gitattributes'
      - 'infection.json.dist'
      - 'phpunit.xml.dist'
      - 'psalm.xml'

File: .github/workflows/build.yml
Comment (vjik) [inline .github/workflows/build.yml:1]: Add `branches: ['master']` on push

File: .github/workflows/mutation.yml
Comment (vjik) [inline .github/workflows/mutation.yml:1]: Add `branches: ['master']` on push

File: README.md [hunk 3]
```diff
@@ -23,36 +23,10 @@ Read more in the [official documentation](https://cycle-orm.dev/docs/readme/2.x)
...
-### Static analysis
-
-The code is statically analyzed with [Psalm](https://psalm.dev/). To run static analysis:
-
-```shell
-./vendor/bin/psalm
-```
+- [English](docs/en/guide)
+- [Russian](docs/ru/guide)
+- [Spanish](docs/es/guide)
```
Comment (vjik) [inline README.md:28]:
```suggestion
- Guide: [English](docs/en/guide), [Russian](docs/ru/guide), [Spanish](docs/es/guide)
```

File: composer.json
Comment (vjik) [inline composer.json:1]: Drop version 4 of psalm

File: rector.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        __DIR__ . '/src',
+        __DIR__ . '/tests',
+    ]);
+
+    // register a single rule
+    $rectorConfig->rule(InlineConstructorDefaultToPropertyRector::class);
+
+    // define sets of rules
+    $rectorConfig->sets([
+        LevelSetList::UP_TO_PHP_80,
```
Comment (vjik) [inline rector.php:21]:
```suggestion
        LevelSetList::UP_TO_PHP_81,
```

File: src/Factory/OrmFactory.php [hunk 5]
```diff
@@ -24,17 +24,15 @@
  */
 final class OrmFactory
 {
-    /**
-     * @psalm-var CollectionsConfig
-     */
-    private array $collectionsConfig;
-
     /**
      * @param CollectionsConfig $collectionsConfig
```
Comment (vjik) [inline src/Factory/OrmFactory.php:28]:
```suggestion
     * @psalm-param CollectionsConfig $collectionsConfig
```

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Factory/OrmFactory.php [hunk 6]
```diff
@@ -24,17 +24,15 @@
...
      */
-    public function __construct(array $collectionsConfig)
-    {
-        $this->collectionsConfig = $collectionsConfig;
+    public function __construct(
+        /**
+         * @psalm-var CollectionsConfig
+         */
+        private array $collectionsConfig
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Factory/OrmFactory.php:34]: Just remove this phpdoc, rector moved it from property.

**Pull request merged**